avm/res/digital-twins/digital-twins-instance/main.bicep

```diff
@@ -0,0 +1,399 @@
+metadata name = 'Digital Twins Instances'
+metadata description = 'This module deploys an Azure Digital Twins Instance.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The name of the Digital Twin Instance.')
+@minLength(3)
+@maxLength(63)
+param name string
+
+@description('Optional. Location for all resources.')
+param location string = resourceGroup().location
+
+@description('Optional. Resource tags.')
+param tags object?
+
+@description('Optional. The lock settings of the service.')
+param lock lockType
+
+@description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+@description('Optional. Event Hub Endpoint.')
+param eventHubEndpoint object = {}
+
+@description('Optional. Event Grid Endpoint.')
+param eventGridEndpoint object = {}
+
+@description('Optional. Service Bus Endpoint.')
+param serviceBusEndpoint object = {}
+
+@description('Optional. Configuration details for private endpoints. For security reasons, it is recommended to use private endpoints whenever possible.')
+param privateEndpoints privateEndpointType
+
+@description('Optional. Whether or not public network access is allowed for this resource. For security reasons it should be disabled. If not specified, it will be disabled by default if private endpoints are set.')
+@allowed([
+  ''
+  'Enabled'
+  'Disabled'
+])
+param publicNetworkAccess string = ''
+
+@description('Optional. The diagnostic settings of the service.')
+param diagnosticSettings diagnosticSettingType
+
+@description('Optional. Enable/Disable usage telemetry for module.')
+param enableTelemetry bool = true
+
+@description('Optional. Array of role assignment objects that contain the \'roleDefinitionIdOrName\' and \'principalIds\' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+param roleAssignments roleAssignmentType
+
+var formattedUserAssignedIdentities = reduce(map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }), {}, (cur, next) => union(cur, next)) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+
+var identity = !empty(managedIdentities) ? {
```

Comment:
Same question as in ` endpoint--event-hub` regarding warning

Reply:
updated

Reply:
some warnings still exist for `identity.` as we discussed previously this is an RP issue